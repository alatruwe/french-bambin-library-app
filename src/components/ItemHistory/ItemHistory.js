import React from "react";
import Item from "../Item/Item";

import data from "../../data.js";

export default class ItemHistory extends React.Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  // API call here to get data

  // API call to delete data
  handleDeleteItem = (event) => {
    event.preventDefault();
    console.log("Item deleted");
    this.props.history.push("./item-history");
  };

  render() {
    return (
      <section className="item-list">
        <h1>Items history</h1>
        <ul>
          {data.map((item) => (
            <li key={item.id}>
              <Item
                title={item.title}
                image={item.img}
                description={item.description}
              />
              <button onClick={this.handleDeleteItem}>Delete item</button>
            </li>
          ))}
        </ul>
      </section>
    );
  }
}
