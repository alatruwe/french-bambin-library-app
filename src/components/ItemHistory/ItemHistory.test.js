import React from "react";
import ReactDOM from "react-dom";
import ItemHistory from "./ItemHistory";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";

describe(`ItemHistory component`, () => {
  it("renders the component by default", () => {
    const wrapper = shallow(<ItemHistory />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
