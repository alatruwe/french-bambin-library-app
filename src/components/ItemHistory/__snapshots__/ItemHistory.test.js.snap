// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemHistory component renders the component by default 1`] = `
<section
  className="item-list"
>
  <h1>
    Items history
  </h1>
  <ul />
</section>
`;
